src/version.json

```diff
@@ -0,0 +1,13 @@
+{
+  "version": "2.7-preview", 
```

Comment:
Once this commit goes in, the version will be calcualted based on version.json;

Reply:
why have -preview there?

Reply:
Because of our nightly builds which publish to myget. We'd publish MSAL 2.6.34-preview, MSAL 2.6.35-preview etc. every night. When we want to publish a release to nuget, we'll have to edit this file to drop the "-preview" bit, then add it back. 

Alternatively we could just publish non-preview packages to myget, since we don't expect people to use them in production (i.e. if they do, we don't support it). This would simplify things as we would never have to edit this file again. 

@MarkZuber , @henrik-me, @jmprieur  - what do you think?

Reply:
We should ideally not have to update a file for this purpose. When we publish bits to nuget/myget I suppose we can mark the bits as preview or pre-release based on input from the build definition?

Reply:
When we push packages to a package source (nuget, myget), there are no such options. The package version controls everything - it is "prerelease" is it has any suffix after the sematic version (e.g. 1.2.3-whatever).

Reply:
then it's an additional manual step? There is really no integration to build configs?

---
In reply to: [244975740](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/749#discussion_r244975740) [](ancestors = 244975740)

Reply:
I prefer the idea of publishing non-preview packages to myget so that we can avoid making mistakes with changing the file for release